lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1901,7 +1901,25 @@ test modules
 =cut
 sub done {
     my ($self, %args) = @_;
-    $args{result} = OBSOLETED if $args{newbuild};
+
+    # read specified result or calculate result from module results if none specified
+    my $result;
+    if ($args{newbuild}) {
+        $result = OBSOLETED;
+    }
+    elsif ($result = $args{result}) {
+        $result = lc($result);
+        die "Erroneous parameters (result invalid)\n" unless grep { $result eq $_ } RESULTS;
+    }
+    else {
+        $result = $self->calculate_result;
+    }
+
+    # limit length of the reason
+    # note: The reason can be anything the worker picked up as useful information so better cut it at a
+    #       reasonable, human-readable length. This also avoids growing the database too big.
+    my $reason = $args{reason};
+    $reason = substr($reason, 0, 120) if defined $reason && length $reason > 120;
```

Comment:
How about adding an ellipsis here to make it visible that the string was indeed cut short?